File: app/src/main/java/upday/droidconmvvm/datamodel/DataModel.java
package upday.droidconmvvm.datamodel;

import androidx.annotation.NonNull;

import java.util.Arrays;
import java.util.List;

import upday.droidconmvvm.model.Language;

public class DataModel implements IDataModel {

    @NonNull
    @Override
    public List<Language> getSupportedLanguages() {
        return Arrays.asList(Language.values());
    }

    @NonNull
    @Override
    public String getGreetingByLanguageCode(@NonNull final Language language) {
        switch (language) {
            case German:
                return "Guten Tag!";
            case English:
                return "Hello!";
            case Slovakian:
                return "Zdravo!";
            case Spanish:
                return "Queobo prro!";
            default:
                return "####";
        }
    }
}

File: app/src/main/java/upday/droidconmvvm/datamodel/DataModel.kt
package upday.droidconmvvm.datamodel

import android.util.Log
import upday.droidconmvvm.model.Language
import java.util.*

class DataModel : IDataModel {
    override fun getSupportedLanguages(): List<Language> = Arrays.asList(*Language.values())

    override fun getGreetingByLanguageCode(language: Language): String =
        when (language) {
            Language.German -> "Guten Tag!"
            Language.English -> "Hello!"
            Language.Slovakian -> "Zdravo!"
            Language.Spanish -> "Queobo prro!"
            else -> "####"
        }.also { Log.d("DataModel", "retrieving greeting") }

}